### PR TITLE
Add unit tests for CustomMenuItemCollection

### PR DESCRIPTION
**Pull request by Philip-Wang01**
Related https://github.com/dotnet/winforms/issues/10773

## Proposed changes
- Add unit test CustomMenuItemCollectionTests.cs for public properties and method of the CustomMenuItemCollection.
- Enable nullability in CustomMenuItemCollection.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12035)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        collection.Count.Should().Be(2);
+    }
+
+    [Fact]
+    public void RefreshItems_ShouldNotThrowException()
+    {
+        CustomMenuItemCollection collection = new();
+
+        Action action = () => collection.RefreshItems();
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs:42]: `() =>`  can be omitted， and please consider test the operation "RefreshItems" after adding some item in the collection

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    [Fact]
+    public void Clear_ShouldClearCollection()
+    {
+        CustomMenuItemCollection collection = new();
+        ToolStripMenuItem[] items =
+        [
+            new ToolStripMenuItem("Item1"),
+            new ToolStripMenuItem("Item2")
+        ];
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs:67]: `ToolStripMenuItem` can be omitted

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        collection.AddRange(items);
+        collection.Clear();
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs:70]: Consider adding an assertion after AddRange to ensure that the Clear operation is actually executed successfully.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace System.Windows.Forms.Design.Tests;
+
+public class CustomMenuItemCollectionTests
+{
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/CustomMenuItemCollectionTests.cs:9]: Maybe avoid instantiating `collection` in every new test method by implementing [xUnit's Constructor and Dispose pattern](https://xunit.net/docs/shared-context#constructor), like so: 

```csharp 
# use this `_collection` field in your tests instead
private readonly CustomMenuItemCollection _collection;

public CustomMenuItemCollectionTests() => _collection = [];

public void Dispose() => _collection.Clear();
```

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Review by lonitra — APPROVED** [review]

**Pull request merged**